xarray/core/combine.py

```diff
@@ -581,6 +582,22 @@ def combine_by_coords(
         temperature     (x) float64 11.04 23.57 20.77 ...
     """
 
+    # Convert dict like objects to Dataset(s)
+    from .dataarray import DataArray
+    from .dataset import Dataset
+
+    dict_like_objects = list()
+    for obj in objects:
```

Comment:
I think this should be a helper function that you can call in `combine_by_coords`, `combine_nested` and `auto_combine`.

Reply:
Thanks! Will do... traveling this week but will try to knock these out.